Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/PointerData/InteractablePointerData.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// pointer information for tracking gestures and pokes
+    /// Abstacts different input types into a single basic set of data
+    /// These values can be accessed from themes or events to
+    /// to provide more granular feedback
+    /// </summary>
+    public struct InteractablePointerTransform
+    {
+        public Vector3 Position;
+        public Quaternion Rotation;
+        public Ray[] Rays;
+    }
+
+    public struct InteractablePressData
+    {
+        public Vector3 StartPosition;
+        public Vector3 Position;
+        public Vector3 Direction;
+        public float Distance;
+        public float Percentage;
+        public bool HasTouch;
+        public float MaxDistance;
+        public Vector3 ProjectedDirection;
+        public Vector3 PressedValue;
+    }
+
+    public class InteractablePointerData
+    {
+        public IMixedRealityPointer Pointer;
+        public HandTrackingInputEventData Hand;
+        public InteractablePointerTransform CurrentTransform;
+        public InteractablePointerTransform StartTransform;
+        public bool HasPress { get; protected set; }
+        public bool HasFocus { get; protected set; }
+        public int ActionScore { get; protected set; }
```

Comment:
What does actionscore mean

Reply:
Early in MRTKv2 I was looking at supporting gestures with hands or controllers. The way the data was structured at the time did not link a MR InputAction to a controller button in the interaction mappings, so there was no way to monitor the controller for the button associated with that action to be released. I believe it's still not as easy as it could be.

I came up with ActionScore as a temporary fix, it could be outdated. A use case was using grip on a controller to pick up an oil can, then while gripping and moving the can around, using trigger to drop oil. We did not want to drop the can prematurely, so action score would store all the values in the controller interaction mappings, at the beginning of an interaction, then wait till the score dropped below that number to handle the release. It's flawed but worked at the time. I need to do another pass here, but was hoping for some guidance.